go/private/rules/cgo.bzl

```diff
@@ -91,14 +91,28 @@ def _select_archive(files):
     for f in files:
       if f.basename.endswith(ext):
         return f
-  fail("cc_library did not produce any files")
+
+def _select_archives(libs):
+  """Selects a one per item in a cc_library, if needed.
+
+  If no archive can be extracted from all the libraries, this will fail.
+  """
+  # list of file extensions in descending order or preference.
+  outs = []
+  for lib in libs:
+    archive = _select_archive(lib.files)
+    if archive:
+      outs.append(archive)
+  if not outs:
+    fail("cc_library(s) did not produce any files")
+  return outs
 
 def _cgo_codegen_impl(ctx):
   go = go_context(ctx)
   if not go.cgo_tools:
     fail("Go toolchain does not support cgo")
   linkopts = ctx.attr.linkopts[:]
-  copts = go.cgo_tools.c_options + go.cgo_tools.compiler_options + ctx.attr.copts
+  cppopts = go.cgo_tools.c_options + go.cgo_tools.compiler_options + ctx.attr.copts + ctx.attr.cppopts
```

Comment:
Let's keep `cppopts` and `copts` separate here for clarity.

```
cppopts = go.cgo_tools.compiler_options +ctx.attr.cppopts
copts = go.cgo_tools.c_options + ctx.attr.copts
```

Reply:
got it